core/bindings.rs

```diff
@@ -467,45 +457,86 @@ fn send(
       })
   });
 
-  let mut buf_one: ZeroCopyBuf;
+  let buf_iter_len = buf_iter.len();
+
+  // Depending on potential errors and the number of args,
+  // this array might only be partially initiallized.
+  // Manual cleanup required, see below.
+  let mut buf_stack: [MaybeUninit<ZeroCopyBuf>; 2] =
+    unsafe { MaybeUninit::uninit().assume_init() };
```

Comment:
This is not a correct way of using `MaybeUninit`.

<sub>In fact it is the *most wrong* way you could possibly use MaybeUninit. Not biggie :) but let's not do this.<sub>

Reply:
I'm inclined to believe you on that, but [the documentation](https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#initializing-an-array-element-by-element) explicitly provides partially initialized arrays as an example.
So I must be doing something wrong; for example maybe taking a slice of the initialized part is invalid for some to me unknown reason?
Could you elaborate on that for me so I don't use it *the most wrong way* again? :)

Reply:
> I'm inclined to believe you on that, but [the documentation](https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#initializing-an-array-element-by-element) explicitly provides partially initialized arrays as an example.

Ouch, that documentation is painfully misleading. I wonder how that got in there.

Anyway -- It's true that you can use it for arrays, but you're really not supposed to call `MaybeUninit::assume_init()` unless you have actually initialized the buffer. The docs do make that very clear:

```rust
// The equivalent code with `MaybeUninit<bool>`:
let b: bool = unsafe { MaybeUninit::uninit().assume_init() }; // undefined behavior!
```

Basically if you call `assume_init()` on the return value of `MaybeUninit::<T>::uninit()` that's never right (unless you are 100% sure that any random combination of bits is a valid `T`, which is true for some primitive values like `u8` and `usize`. But definitely not for any references nor `bool` nor `NonNull` etc.

 So if you want to use it with an array then you should do:

```
// Create an uninitialized slice.
let mut buf_stack: [MaybeUninit<Foo>; 2] = unsafe { MaybeUninit::uninit() };

// Initialize!
unsafe { std::ptr::write(buf_stack[0].as_ptr(), Foo::new()) };

// Not you can safely call assume_init (or just cast to an initialized array):
let slice_of_1 = unsafe { &*(&buf_stack as *const _ as *const [Foo; 1]) };
```



let arr 

```rust

Reply:
I see, thanks for your explanation.
`[MaybeUninit::uninit(); 2]` wouldn't work yet, but `MaybeUninit::uninit_array()` would be it (for nightly).